taskrun/TaskManager.py

```diff
@@ -308,13 +284,61 @@ def _task_error(self, task, errors):
       # clean up the task
       self._task_done(task)
 
+  def _kill_running(self, task):
+    """
+    Kills all running tasks except for the specified task.
```

Comment:
I don't think the "except for the specified task" is necessary any more, since task.kill() is now idempotent

Reply:
What about the _kill_running(None) case?

Reply:
I mean it can just call kill() on all the tasks and doesn't need a separate argument.

Reply:
Ok, I've changed this but it required changes in the *Task.py files. The change is to block the setting of self.killed = True if the task has already run. This is needed for proper post-mortem information about which failed and which were killed.